pkg/health/health.go

```diff
@@ -68,8 +76,17 @@ func (h ConsulHealthChecker) WatchNodeService(nodename string, serviceID string,
 					catalogResults = append(catalogResults, outResult)
 				}
 				kvCheckResult := consulWatchToResult(kvCheck)
-				catalogCheckResult := findWorstResult(catalogResults)
-				resultCh <- findBestResult([]Result{kvCheckResult, catalogCheckResult})
+				catalogCheckResult, HEErr := findWorstResult(catalogResults)
+				if HEErr != nil {
+					resultCh <- kvCheckResult
```

Comment:
You need to check if `kvCheckResult` is actually present since it could be empty, right?